tests/phpunit/CRM/Upgrade/Incremental/php/FiveTwentyTest.php

```diff
@@ -0,0 +1,198 @@
+<?php
+require_once 'CiviTest/CiviCaseTestCase.php';
+
+/**
+ * Class CRM_Upgrade_Incremental_php_FiveTwentyTest
+ * @group headless
+ */
+class CRM_Upgrade_Incremental_php_FiveTwentyTest extends CiviCaseTestCase {
+
+  public function setUp() {
+    parent::setUp();
+  }
+
+  /**
+   * Test that the upgrade task changes the direction but only
+   * for bidirectional relationship types that are b_a.
+   */
+  public function testChangeCaseTypeAutoassignee() {
+
+    // We don't know what the ids are for the relationship types since it
+    // seems to depend what ran before us, so retrieve them first and go by
+    // name.
+    // Also spouse might not exist.
+
+    $result = $this->callAPISuccess('RelationshipType', 'get', ['limit' => 0])['values'];
+    // Get list of ids keyed on name.
+    $relationshipTypeNames = array_column($result, 'id', 'name_b_a');
+
+    // Create spouse if none.
+    if (!isset($relationshipTypeNames['Spouse of'])) {
+      $spouseId = $this->relationshipTypeCreate([
+        'name_a_b' => 'Spouse of',
+        'name_b_a' => 'Spouse of',
+        'label_a_b' => 'Spouse of',
+        'label_b_a' => 'Spouse of',
+        'contact_type_a' => 'Individual',
+        'contact_type_b' => 'Individual',
+      ]);
+      $relationshipTypeNames['Spouse of'] = $spouseId;
+    }
+    // Maybe unnecessary but why not. Slightly different than an undefined
+    // index later when it doesn't exist at all.
+    $this->assertGreaterThan(0, $relationshipTypeNames['Spouse of']);
+
+    /**
+     * Set up xml. Can't use complex variable substitution in heredocs so
+     * have some %% placeholders and we'll replace after.
+     * In this sample case type for autoassignees we have:
+     * - a_b unidirectional: Case Coordinator is
+     * - b_a unidirectional: Benefits Specialist
+     * - Bidirectional the way it's stored in 5.16+: Spouse of
+     * - config entry pre-5.16, where it's a bidirectional relationship stored as b_a: Spouse of
+     */
+    $newCaseTypeXml = <<<ENDXML
```

Comment:
Be still my heart--you're a fan of HEREDOC *and* Markdown tables!

Reply:
Hear hear! (not sure if the pun translates well...)
Although as I look at it now this one might have made more sense as a double-quoted string and then I wouldn't need the substitution loop after it I could just use the array variables inside the string.

Reply:
No that's what's so nice about the HEREDOC--you *can* use the variables in the string.  It's just like the double quotes except you don't even have to escape double quotes.  [NOWDOC](https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.nowdoc) is the one that acts like the single quotes.

Reply:
When I tried it though it only worked with simple vars like `$foo`, not with `{$foo['bar']}`. But I guess instead of the loop, since there's only 3, I could have made simple var aliases and then used those in the heredoc.

Reply:
Of course when I try it now it works... maybe I made a typo the first time around.